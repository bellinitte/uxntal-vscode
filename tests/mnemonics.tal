BRK JMP JMP2 JMPk JMPr JMP2kr2kr ( Valid mnemonics )

( Invalid modes )
BRK2 BRKk BRKr
JCI2 JCIk JCIr
JMI2 JMIk JMIr
JSI2 JSIk JSIr

aBRK BRKa BRK- BRK-JMP JMP2kr2jr JMPa JMP- ( Almost-mnemonics )

( All base mnemonics )
BRK JCI JMI JSI
LIT INC POP NIP SWP ROT DUP OVR
EQU NEQ GTH LTH JMP JCN JSR STH
LDZ STZ LDR STR LDA STA DEI DEO
ADD SUB MUL DIV AND ORA EOR SFT
