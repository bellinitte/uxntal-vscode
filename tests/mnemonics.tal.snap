>BRK JMP JMP2 JMPk JMPr JMP2kr2kr ( Valid mnemonics )
#^^^ source.tal keyword.uxntal
#   ^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^^ source.tal keyword.uxntal
#            ^ source.tal
#             ^^^^ source.tal keyword.uxntal
#                 ^ source.tal
#                  ^^^^ source.tal keyword.uxntal
#                      ^ source.tal
#                       ^^^^^^^^^ source.tal keyword.uxntal
#                                ^ source.tal
#                                 ^ source.tal comment.uxntal
#                                  ^ source.tal
#                                   ^^^^^ source.tal comment.uxntal
#                                        ^ source.tal
#                                         ^^^^^^^^^ source.tal comment.uxntal
#                                                  ^ source.tal
#                                                   ^ source.tal comment.uxntal
>
>( Invalid modes )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^^ source.tal comment.uxntal
#         ^ source.tal
#          ^^^^^ source.tal comment.uxntal
#               ^ source.tal
#                ^ source.tal comment.uxntal
>BRK2 BRKk BRKr
#^^^^ source.tal entity.name.function.uxntal
#    ^ source.tal
#     ^^^^ source.tal entity.name.function.uxntal
#         ^ source.tal
#          ^^^^ source.tal entity.name.function.uxntal
>JCI2 JCIk JCIr
#^^^^ source.tal entity.name.function.uxntal
#    ^ source.tal
#     ^^^^ source.tal entity.name.function.uxntal
#         ^ source.tal
#          ^^^^ source.tal entity.name.function.uxntal
>JMI2 JMIk JMIr
#^^^^ source.tal entity.name.function.uxntal
#    ^ source.tal
#     ^^^^ source.tal entity.name.function.uxntal
#         ^ source.tal
#          ^^^^ source.tal entity.name.function.uxntal
>JSI2 JSIk JSIr
#^^^^ source.tal entity.name.function.uxntal
#    ^ source.tal
#     ^^^^ source.tal entity.name.function.uxntal
#         ^ source.tal
#          ^^^^ source.tal entity.name.function.uxntal
>
>aBRK BRKa BRK- BRK-JMP JMP2kr2jr JMPa JMP- ( Almost-mnemonics )
#^^^^ source.tal entity.name.function.uxntal
#    ^ source.tal
#     ^^^^ source.tal entity.name.function.uxntal
#         ^ source.tal
#          ^^^^ source.tal entity.name.function.uxntal
#              ^ source.tal
#               ^^^^^^^ source.tal entity.name.function.uxntal
#                      ^ source.tal
#                       ^^^^^^^^^ source.tal entity.name.function.uxntal
#                                ^ source.tal
#                                 ^^^^ source.tal entity.name.function.uxntal
#                                     ^ source.tal
#                                      ^^^^ source.tal entity.name.function.uxntal
#                                          ^ source.tal
#                                           ^ source.tal comment.uxntal
#                                            ^ source.tal
#                                             ^^^^^^^^^^^^^^^^ source.tal comment.uxntal
#                                                             ^ source.tal
#                                                              ^ source.tal comment.uxntal
>
>( All base mnemonics )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^ source.tal comment.uxntal
#     ^ source.tal
#      ^^^^ source.tal comment.uxntal
#          ^ source.tal
#           ^^^^^^^^^ source.tal comment.uxntal
#                    ^ source.tal
#                     ^ source.tal comment.uxntal
>BRK JCI JMI JSI
#^^^ source.tal keyword.uxntal
#   ^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^ source.tal keyword.uxntal
#           ^ source.tal
#            ^^^ source.tal keyword.uxntal
>LIT INC POP NIP SWP ROT DUP OVR
#^^^ source.tal keyword.uxntal
#   ^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^ source.tal keyword.uxntal
#           ^ source.tal
#            ^^^ source.tal keyword.uxntal
#               ^ source.tal
#                ^^^ source.tal keyword.uxntal
#                   ^ source.tal
#                    ^^^ source.tal keyword.uxntal
#                       ^ source.tal
#                        ^^^ source.tal keyword.uxntal
#                           ^ source.tal
#                            ^^^ source.tal keyword.uxntal
>EQU NEQ GTH LTH JMP JCN JSR STH
#^^^ source.tal keyword.uxntal
#   ^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^ source.tal keyword.uxntal
#           ^ source.tal
#            ^^^ source.tal keyword.uxntal
#               ^ source.tal
#                ^^^ source.tal keyword.uxntal
#                   ^ source.tal
#                    ^^^ source.tal keyword.uxntal
#                       ^ source.tal
#                        ^^^ source.tal keyword.uxntal
#                           ^ source.tal
#                            ^^^ source.tal keyword.uxntal
>LDZ STZ LDR STR LDA STA DEI DEO
#^^^ source.tal keyword.uxntal
#   ^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^ source.tal keyword.uxntal
#           ^ source.tal
#            ^^^ source.tal keyword.uxntal
#               ^ source.tal
#                ^^^ source.tal keyword.uxntal
#                   ^ source.tal
#                    ^^^ source.tal keyword.uxntal
#                       ^ source.tal
#                        ^^^ source.tal keyword.uxntal
#                           ^ source.tal
#                            ^^^ source.tal keyword.uxntal
>ADD SUB MUL DIV AND ORA EOR SFT
#^^^ source.tal keyword.uxntal
#   ^ source.tal
#    ^^^ source.tal keyword.uxntal
#       ^ source.tal
#        ^^^ source.tal keyword.uxntal
#           ^ source.tal
#            ^^^ source.tal keyword.uxntal
#               ^ source.tal
#                ^^^ source.tal keyword.uxntal
#                   ^ source.tal
#                    ^^^ source.tal keyword.uxntal
#                       ^ source.tal
#                        ^^^ source.tal keyword.uxntal
#                           ^ source.tal
#                            ^^^ source.tal keyword.uxntal
>